src/drivers/spektrum_rc/spektrum_rc.cpp

```diff
@@ -0,0 +1,304 @@
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+
+/**
+ * @file spektrum_rc.cpp
+ *
+ * This is a driver for a Spektrum satellite receiver connected to a Snapdragon
+ * on the serial port. By default port J15 (next to USB) is used.
+ */
+
+#include <px4_tasks.h>
+#include <px4_posix.h>
+
+#include <lib/rc/dsm.h>
+#include <drivers/drv_rc_input.h>
+#include <drivers/drv_hrt.h>
+
+#include <uORB/uORB.h>
+#include <uORB/topics/input_rc.h>
+
+// Snapdraogon: use J15 (next to USB)
+#define SPEKTRUM_UART_DEVICE_PATH "/dev/tty-1"
+
+#define UNUSED(x) (void)(x)
+
+extern "C" { __EXPORT int spektrum_rc_main(int argc, char *argv[]); }
+
+
+namespace spektrum_rc
+{
+
+volatile bool _task_should_exit = false;
+static bool _is_running = false;
+static px4_task_t _task_handle = -1;
+
+int start();
+int stop();
+int info();
+void usage();
+void task_main(int argc, char *argv[]);
+
+void fill_input_rc(uint16_t raw_rc_count, uint16_t raw_rc_values[input_rc_s::RC_INPUT_MAX_CHANNELS],
+		   hrt_abstime now, bool frame_drop, bool failsafe, unsigned frame_drops, int rssi,
+		   input_rc_s &input_rc);
+
+void task_main(int argc, char *argv[])
+{
+	int uart_fd = dsm_init(SPEKTRUM_UART_DEVICE_PATH);
+
+	if (uart_fd < 1) {
+		PX4_ERR("dsm init failed");
+		return;
+	}
+
+	orb_advert_t rc_pub = nullptr;
+
+	// Use a buffer size of the double of the minimum, just to be safe.
+	uint8_t rx_buf[2 * DSM_BUFFER_SIZE];
+
+	// Set up poll topic
+	px4_pollfd_struct_t fds[1];
+	fds[0].fd     = uart_fd;
+	fds[0].events = POLLIN;
+
+	_is_running = true;
+
+	// Main loop
+	while (!_task_should_exit) {
+
+		int pret = px4_poll(&fds[0], (sizeof(fds) / sizeof(fds[0])), 50);
+
+		/* Timed out, do a periodic check for _task_should_exit. */
+		if (pret == 0) {
+			continue;
+		}
+
+		if (pret < 0) {
+			PX4_WARN("poll error %d, %d", pret, errno);
+			/* sleep a bit before next try */
+			usleep(100000);
+			continue;
+		}
+
+		if (!(fds[0].revents & POLLIN)) {
+			// There is only one fd in poll, so this should be impossible.
+			PX4_WARN("poll bit not set");
+			continue;
+		}
+
+		int newbytes = ::read(uart_fd, &rx_buf[0], sizeof(rx_buf));
+
+		if (newbytes < 0) {
+			PX4_WARN("read failed");
+			continue;
+		}
+
+		if (newbytes == 0) {
+			PX4_WARN("read nothing");
+			continue;
+		}
+
+
+		const hrt_abstime now = hrt_absolute_time();
+
+		bool dsm_11_bit;
+		unsigned frame_drops;
+		uint16_t raw_rc_values[input_rc_s::RC_INPUT_MAX_CHANNELS];
+		uint16_t raw_rc_count;
+
+		// parse new data
+		bool rc_updated = dsm_parse(now, rx_buf, newbytes, &raw_rc_values[0], &raw_rc_count,
+					    &dsm_11_bit, &frame_drops, input_rc_s::RC_INPUT_MAX_CHANNELS);
```

Comment:
Same here. This notion is more explicit and more prone to result in (correct) compile errors if the variable would not hold an array but something else.